### PR TITLE
Merge Branch-0.7

### PR DESCRIPTION
**Pull request by calvinjia**
These are all the changes that were made to the 0.7 branch that should be ported to master as well.

### CODE REVIEW

File: docs/Tiered-Storage-on-Tachyon.md [hunk 1]
```diff
@@ -1,90 +1,208 @@
...
+<tr>
+  <td>tachyon.worker.allocate.strategy</td>
+  <td>MAX_FREE</td>
+  <td>
+  The allocation strategy to use for new blocks in Tachyon. Currently, the possible strategies are
+  GREEDY and MAX_FREE.
+  </td>
+</tr>
+<tr>
+  <td>tachyon.worker.evict.strategy</td>
```
Comment (gpang) [inline docs/Tiered-Storage-on-Tachyon.md]: This parameter is now *.class and with the class name.

File: docs/Tiered-Storage-on-Tachyon.md [hunk 2]
```diff
@@ -1,90 +1,208 @@
...
+  <td>128MB (for tachyon.worker.tieredstore.level0.dirs.quota)</td>
+  <td>
+  The quotas for all storage directories in storage tier x, delimited by comma. x represents the
+  storage tier number (starting from 0). For a particular storage tier, if the list of quotas is
+  shorter than the list of directories of that tier, then the quotas for the remaining directories
+  will just use the last-defined quota. Quota definitions use these suffixes: KB, MB, GB, TB, PB.
+  </td>
+</tr>
+<tr>
+  <td>tachyon.worker.allocate.strategy</td>
```
Comment (gpang) [inline docs/Tiered-Storage-on-Tachyon.md]: This parameter is now *.class and with the class name.

File: docs/Tiered-Storage-on-Tachyon.md [hunk 3]
```diff
@@ -1,90 +1,208 @@
...
+
+There are a few restrictions to defining the tiers. First of all, there can be at most 3 tiers.
+Also, at most 1 tier can refer to a specific alias. For example, at most 1 tier can have the alias
+HDD. If you want Tachyon to use multiple hard drives for the HDD tier, you can configure that as
+multiple storage directories in the HDD tier.
+
+Additionally, the specific evictor and allocator strategies can be configured. Those configuration
+parameters are:
+
+    tachyon.worker.allocate.strategy=MAX_FREE
```
Comment (gpang) [inline docs/Tiered-Storage-on-Tachyon.md]: I think master branch now changed these (this one and the one below) to be "tachyon.worker.allocate.*.class", and with the class name.

Comment (apc999) [inline docs/Tiered-Storage-on-Tachyon.md]: tachyon.worker.allocate.strategy.class=tachyon.worker.block.allocator.MaxFreeAllocator

File: docs/Tiered-Storage-on-Tachyon.md (hunk 1 above)
Comment (calvinjia) [inline docs/Tiered-Storage-on-Tachyon.md]: Fixed, thanks.

File: docs/Tiered-Storage-on-Tachyon.md (hunk 2 above)
Comment (calvinjia) [inline docs/Tiered-Storage-on-Tachyon.md]: Fixed, thanks.

File: docs/Tiered-Storage-on-Tachyon.md (hunk 3 above)
Comment (calvinjia) [inline docs/Tiered-Storage-on-Tachyon.md]: Fixed, thanks.

File: docs/Tiered-Storage-on-Tachyon.md [hunk 4]
```diff
@@ -1,90 +1,208 @@
...
+tier, then the evictor is triggered in order to free space for the new block.
+
+## Reading Data
+
+Reading a data block with tiered storage is similar to standard Tachyon. Tachyon will simply read
+the block from where it is already stored. If Tachyon is configured with multiple tiers, then the
+block will not be necessarily read from the MEM tier, since it could have been moved to a lower tier
+transparently.
+
+### Promoting Blocks
```
Comment (uronce-cc) [inline docs/Tiered-Storage-on-Tachyon.md]: `###` seems like a subsection of `## Reading Data`?

File: docs/Tiered-Storage-on-Tachyon.md [hunk 5]
```diff
@@ -1,90 +1,208 @@
...
+  shorter than the list of directories of that tier, then the quotas for the remaining directories
+  will just use the last-defined quota. Quota definitions use these suffixes: KB, MB, GB, TB, PB.
+  </td>
+</tr>
+<tr>
+  <td>tachyon.worker.allocate.strategy.class</td>
+  <td>tachyon.worker.block.allocator.MaxFreeAllocator</td>
+  <td>
+  The class name of the allocation strategy to use for new blocks in Tachyon. Currently, the
+  possible strategies are GREEDY and MAX_FREE.
```
Comment (uronce-cc) [inline docs/Tiered-Storage-on-Tachyon.md]: What about replacing `GREEDY` and `MAX_FREE` with `tachyon.worker.block.allocator.GreedyAllocator` and `tachyon.worker.block.allocator.MaxFreeAllocator`? Seems more direct for users to know other options of allocators.

File: docs/Tiered-Storage-on-Tachyon.md [hunk 6]
```diff
@@ -1,90 +1,208 @@
...
+  </td>
+</tr>
+<tr>
+  <td>tachyon.worker.evict.strategy.class</td>
+  <td>tachyon.worker.block.evictor.LRUEvictor</td>
+  <td>
+  The class name of the block eviction strategy to use when a storage layer runs out of space.
+  Currently, the supported strategies are GREEDY and LRU.
```
Comment (uronce-cc) [inline docs/Tiered-Storage-on-Tachyon.md]: same as above.

File: docs/Tiered-Storage-on-Tachyon.md (hunk 4 above)
Comment (calvinjia) [inline docs/Tiered-Storage-on-Tachyon.md]: Let's make logical changes to in another PR? This is just to merge the work in 0.7 back, but the compatibility changes can be made here as kind of a merge conflict resolution.

File: docs/Tiered-Storage-on-Tachyon.md (hunk 5 above)
Comment (calvinjia) [inline docs/Tiered-Storage-on-Tachyon.md]: Sure, at first I thought it would be too verbose, but `GREEDY` and `MAX_FREE` no longer have meaning in the codebase.

File: docs/Tiered-Storage-on-Tachyon.md [hunk 7]
```diff
@@ -1,90 +1,208 @@
...
+  </td>
+</tr>
+</table>
```
Comment (uronce-cc) [inline docs/Tiered-Storage-on-Tachyon.md]: Could we merge this explanation of configuration parameters to https://github.com/amplab/tachyon/blob/master/docs/Configuration-Settings.md or tell users about the explanation here in the `Configuration-Settings.md`? Users may not find it.

File: docs/Tiered-Storage-on-Tachyon.md (hunk 6 above)
Comment (calvinjia) [inline docs/Tiered-Storage-on-Tachyon.md]: Updated, thanks.

**Comment by uronce-cc** [comment]
+1 LGTM

**Comment by apc999** [comment]
+1 LGTM

**Comment by gpang** [comment]
LGTM
